8Knot/cache_manager/db_init.py

```diff
@@ -117,11 +117,12 @@ def _create_application_tables() -> None:
 
     with conn.cursor() as cur:
         # create tables if they don't already exist.
+        # TODO: id->repo_id, commits->commit_id
         cur.execute(
             """
             CREATE UNLOGGED TABLE IF NOT EXISTS commits_query(
-                id integer,
-                commits text PRIMARY KEY,
+                id int,
+                commits text, -- this is the commit hash, so it's base64 hash.
```

Comment:
what does this mean exactly (not a needed change, just want to make sure I understand)

Reply:
Every commit has a hash value, which is unique. In the augur commits table, multiple rows have the same hash, referring to the same commit, but to different files in the commit. the hash is in 'base64' which means its a string that looks like "012a1bbed8a9092dec29fff" or something, which is a more compact form for representing binary. 

so the datatype is text, because utf-8 characters are being used to represent many binary digits.